src/lib/libmints/deriv.cc

```diff
@@ -747,14 +751,16 @@ SharedMatrix Deriv::compute()
     corr->add(x_contr_);
     corr->add(tpdm_contr_);
     if (reference_separate) {
-        gradient_->add(x_ref_contr_);
-        gradient_->add(opdm_ref_contr_);
-        gradient_->add(tpdm_ref_contr_);
-        SharedMatrix scf_gradient(gradient_->clone());
-        scf_gradient->set_name("Reference Gradient");
-        scf_gradient->print_atom_vector();
-        wfn_->reference_wavefunction()->set_gradient(scf_gradient);
-        corr->print_atom_vector();
+        if ( x_ref_contr_ && opdm_ref_contr_ && tpdm_ref_contr_) {
```

Comment:
Hmm, this logic scares me a bit. Id rather make sure the `reference_separate` is meaningful instead.

Looks like this is set by `bool reference_separate = (X) && ref_wfn;`. The `reference_wavefunction_` attribute must be set to _something_ in whatever you passed into Deriv.

@jturney, @andysim  This has always confused me a bit. What was the original goal of `reference_wavefunction_`?